packages/o-spreadsheet-engine/src/helpers/figures/charts/runtime/chart_highlight.ts

```diff
@@ -0,0 +1,123 @@
+import { ChartDataset } from "chart.js";
+import { Color } from "../../../..";
+import { ChartType } from "../../../../types/chart";
+import { setColorAlpha } from "../../../color";
+
+export function getHighlightsHelpers(type: ChartType) {
+  switch (type) {
+    case "scatter":
+    case "radar":
+    case "line":
+      return {
+        highlightItem: highlightLineChartItem,
+        unHighlightItems: unHighlightLineChartItems,
+      };
+    case "bar":
+      return {
+        highlightItem: highlightBarChartItem,
+        unHighlightItems: unHighlightBarChartItems,
+      };
+    case "pie":
+      return {
+        highlightItem: highlightPieChartItem,
+        unHighlightItems: unHighlightPieChartItems,
+      };
+    case "combo":
+      return {
+        highlightItem: highlightComboChartItem,
+        unHighlightItems: unHighlightComboChartItems,
+      };
+    default:
+      return {
+        highlightItem: () => {},
+        unHighlightItems: () => {},
+      };
+  }
+}
+
+function highlightPieChartItem(item: { index?: number }, dataSets: ChartDataset[]) {
+  const index = item.index;
+  dataSets.forEach((dataset) => {
+    const backgroundColors = dataset.backgroundColor as Color[] | undefined;
+    if (!backgroundColors) {
+      return;
+    }
+    backgroundColors.forEach((color, i, colors) => {
+      colors[i] = setColorAlpha(color, i === index ? 1 : 0.4);
+    });
+  });
+}
+
+function highlightLineChartItem(item: { datasetIndex?: number }, dataSets: ChartDataset[]) {
+  const index = item.datasetIndex;
+  dataSets.forEach((dataset, i) => {
+    const color = setColorAlpha(dataset.borderColor as Color, i === index ? 1 : 0.4);
```

Comment:
ooh I think I remember one of the reason I had a hard time with sunburst. `dataset.borderColor` can be a callback, in which case this doesn't work. I don't think we ever use callbacks for pie/bar/line so it should be good here